broker/aws/impl.go

```diff
@@ -598,3 +621,23 @@ func (b *Broker) generateTokenCredentials(accountName string, roleName string, u
 	b.auditLogger.Printf("Token credentials generated for: %s on account %s role %s", userName, accountName, roleName)
 	return &outVal, nil
 }
+
+func (b *Broker) updateConfiguration(
+	config *configuration.Configuration) error {
+	if config == nil {
+		return errors.New("nill config passed")
+	}
+	if config.AWS.GroupPrefix == "" {
+		b.userInfo = b.rawUserInfo
+	} else {
+		ui, err := filter.NewUserGroupsFilter(b.rawUserInfo,
```

Comment:
in the case of having a GroupPrefix.. wont this make a new filter wrapping the filtered data already?

Reply:
`rawUserInfo` does not have a group prefix filter already. I've abstracted that functionality into a common package so that the low-level implementations don't have to duplicate that.